e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,612 @@
+	// EXPECTED: Request fails; Minter Controller remains configured with Minter
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "alice", math.OneInt(), noble)
+	alice := w[0]
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "configure-minter-controller", nw.fiatTfRoles.MinterController.FormattedAddress(), nw.fiatTfRoles.Minter.FormattedAddress())
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err = showMinterController(ctx, val, nw.fiatTfRoles.MinterController)
+	require.NoError(t, err, "failed to query show-minter-controller")
+
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "remove-minter-controller", nw.fiatTfRoles.MinterController.FormattedAddress())
+	require.ErrorContains(t, err, "you are not the master minter: unauthorized")
+
+	showMCRes, err = showMinterController(ctx, val, nw.fiatTfRoles.MinterController)
+	require.NoError(t, err, "failed to query show-minter-controller")
+
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	// ACTION: Remove Minter Controller while Minter and Minter Controller are blacklisted
+	// EXPECTED: Success; Minter Controller is removed
+	// Status:
+	// 	gw minterController -> gw minter
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.MasterMinter)
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.MinterController)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "remove-minter-controller", nw.fiatTfRoles.MinterController.FormattedAddress())
+	require.NoError(t, err, "error removing minter controller")
+
+	_, err = showMinterController(ctx, val, nw.fiatTfRoles.MinterController)
+	require.Error(t, err, "successfully queried for the minter controller when it should have failed")
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.MasterMinter)
+
+	// ACTION: Remove a a non existent Minter Controller
+	// EXPECTED: Request fails
+	// Status:
+	// 	no minterController setup
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "remove-minter-controller", nw.fiatTfRoles.MinterController.FormattedAddress())
+	require.ErrorContains(t, err, fmt.Sprintf("minter controller with a given address (%s) doesn't exist: user not found", nw.fiatTfRoles.MinterController.FormattedAddress()))
+
+}
+
+func TestFiatTFConfigureMinter(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Happy path: Configure minter
+	// EXPECTED: Success; Minter is configured with allowance
+
+	configureMinter(t, ctx, val, nw.fiatTfRoles.MinterController, nw.fiatTfRoles.Minter, 20)
```

Comment:
missing: verifying the minter allowance was set properly.

Reply:
Verification is done inside the `configureMinter` function. See: https://github.com/noble-assets/noble/blob/9143e5ca0edfa81e35da6fe9edfdb92845779ca9/e2e/utils.go#L483-L499